src/routes/safe/components/Apps/hooks/useIframeMessageHandler.ts

```diff
@@ -104,6 +101,37 @@ const useIframeMessageHandler = (
           break
         }
 
+        case SDK_MESSAGES.RPC_CALL: {
+          const payload = messagePayload as SDKMessageToPayload['RPC_CALL']
+
+          if (
+            web3ReadOnly &&
+            web3ReadOnly.currentProvider != null &&
+            typeof web3ReadOnly.currentProvider !== 'string' &&
+            'send' in web3ReadOnly.currentProvider
+          ) {
+            web3ReadOnly.currentProvider?.send?.(
```

Comment:
why the optional chaining operator for `currentProvider`?

Reply:
If I remove it typescript complains 🙈

Reply:
reviewing it a bit more, wouldn't be the same to do `web3ReadOnly?.currentProvider?.send?.(` and avoid the `if` conditions?

Reply:
<img width="1305" alt="Screenshot 2020-10-23 at 12 45 52" src="https://user-images.githubusercontent.com/16622558/96977253-e1be7880-152d-11eb-8d4b-96b19b2b35d5.png">

Reply:
Yesss, I was thinking about it and I was about to say that that suggestion was lame. Sorry

BTW, the `break` is intentionally left inside the `if` block?

Reply:
No worries! No, that was unintentional, I moved it out in my latest commit